src/ProjectSystemTools/BuildLogging/Model/Frontend/FrontendBuildTableDataSource.cs

```diff
@@ -3,39 +3,36 @@
 using System;
 using System.Collections.Immutable;
 using System.ComponentModel.Composition;
-using Microsoft.Build.Evaluation;
-using Microsoft.Build.Framework;
+using Microsoft.VisualStudio.ProjectSystem.Tools.BuildLogging.Model.RpcContracts;
 using Microsoft.VisualStudio.ProjectSystem.Tools.BuildLogging.UI;
 using Microsoft.VisualStudio.ProjectSystem.Tools.Providers;
 using Microsoft.VisualStudio.Shell.TableManager;
 
-namespace Microsoft.VisualStudio.ProjectSystem.Tools.BuildLogging.Model
+namespace Microsoft.VisualStudio.ProjectSystem.Tools.BuildLogging.Model.Frontend
 {
-    [Export(typeof(IBuildTableDataSource))]
-    internal sealed class BuildTableDataSource : ITableEntriesSnapshotFactory, IBuildTableDataSource
+    [Export(typeof(IFrontendBuildTableDataSource))]
+    internal sealed class FrontendBuildTableDataSource : ITableEntriesSnapshotFactory, IFrontendBuildTableDataSource
     {
         private const string BuildDataSourceDisplayName = "Build Data Source";
         private const string BuildTableDataSourceIdentifier = nameof(BuildTableDataSourceIdentifier);
         private const string BuildTableDataSourceSourceTypeIdentifier = nameof(BuildTableDataSourceSourceTypeIdentifier);
 
         private readonly object _gate = new object();
-        private readonly EvaluationLogger _evaluationLogger;
-        private readonly RoslynLogger _roslynLogger;
 
         private ITableManager _manager;
         private ITableDataSink _tableDataSink;
         private BuildTableEntriesSnapshot _lastSnapshot;
-        private ImmutableList<Build> _entries = ImmutableList<Build>.Empty;
+        private ImmutableList<UIBuildSummary> _entries = ImmutableList<UIBuildSummary>.Empty;
```

Comment:
(This comment is not about your change, so feel free to ignore.)

Looking at the usage of `_entries` in this class, I think `ImmutableArray<>` would be a better fit than `ImmutableList<>` from a performance point of view. We can chat about this next week if you like.